src/main/scala/viper/gobra/frontend/info/implementation/typing/IdTyping.scala

```diff
@@ -28,12 +29,26 @@ trait IdTyping extends BaseTyping { this: TypeInfoImpl =>
     }
   }
 
-  private[typing] def wellDefActualRegular(entity: ActualRegular, id: PIdnNode): ValidityMessages = entity match {
+  /**
+    * Returns true if n or any of its children (in the current package) have an entity that corresponds to one in the set e
+    */
+  private def isNotCyclic(n: PNode, e: Set[Entity]): ValidityMessages = LocalMessages(message(n, s"got cyclic structure starting at $n", n match {
+    case n: PIdnNode => entity(n) match {
+      case r if e.contains(r) => true
+      case SingleConstant(_, _, _, _, _, ctx) if this != ctx => false // we do not follow the evaluation into different packages
```

Comment:
Why do we not follow the evaluation into another package? I guess that it is not necessary because the other package cannot point back into the current package because imports are not allowed to have a cycle.

Reply:
yes exactly, I've extended the comment